File: le-serviceflows/core/src/main/resources/com/latticeengines/serviceflows/core/transforms/std_visidb_ds_title_isacademic.py

#
# $LastChangedBy$
# $LastChangedDate$
# $Rev$
#

import re
from std_visidb_ds_title_level import std_visidb_ds_title_level

def metadata():
    return {  'ApprovedUsage'   : 'ModelAndAllInsights'
            , 'Description'     : 'Indicator for Academic Job Title'
            , 'DisplayName'     : 'Has Academic Title'
            , 'Category'        : 'Lead Information'
            , 'FundamentalType' : 'boolean'
            , 'StatisticalType' : 'nominal'
            , 'Tags'            : 'InternalTransform' }


def std_visidb_ds_title_isacademic(n):
    if n is None:
        return False

    if re.search('student|researcher|professor|dev|programmer', n.lower()) and std_visidb_ds_title_level(n) == 0:
        return True

    return False


def transform(args, record):
    column = args["column"]
    value = record[column]
    return std_visidb_ds_title_isacademic(value)

File: le-transform/src/main/java/com/latticeengines/transform/v2_0_25/functions/StdVisidbDsTitleIsacademic.java
package com.latticeengines.transform.v2_0_25.functions;

import java.util.Map;
import java.util.regex.Pattern;

import com.latticeengines.transform.exposed.RealTimeTransform;
import com.latticeengines.transform.exposed.metadata.ApprovedUsage;
import com.latticeengines.transform.exposed.metadata.Category;
import com.latticeengines.transform.exposed.metadata.FundamentalType;
import com.latticeengines.transform.exposed.metadata.StatisticalType;
import com.latticeengines.transform.exposed.metadata.Tag;
import com.latticeengines.transform.exposed.metadata.TransformMetadata;

public class StdVisidbDsTitleIsacademic implements RealTimeTransform {

    private static final long serialVersionUID = -3175023468680385610L;

    public StdVisidbDsTitleIsacademic() {
    }

    public StdVisidbDsTitleIsacademic(String modelPath) {
    }

    public Object transform(Map<String, Object> arguments, Map<String, Object> record) {
        String column = (String) arguments.get("column");
        Object n = record.get(column);

        if (n == null)
            return false;

        String s = n.toString().toLowerCase();

        return (Pattern.matches(
                "(.*?\\b)student(.*)|(.*?\\b)researcher(.*)|(.*?\\b)professor(.*)|(.*)dev(.*)|(.*?\\b)programmer(.*)",
                s) && StdVisidbDsTitleLevel.calculateStdVisidbDsTitleLevel(s) == 0) ? true : false;
    }

    @Override
    public TransformMetadata getMetadata() {
        TransformMetadata metadata = new TransformMetadata();
        metadata.setApprovedUsage(ApprovedUsage.MODEL_ALLINSIGHTS);
        metadata.setCategory(Category.LEAD_INFORMATION);
        metadata.setFundamentalType(FundamentalType.BOOLEAN);
        metadata.setStatisticalType(StatisticalType.NOMINAL);
        metadata.setTags(Tag.INTERNAL_TRANSFORM);
        metadata.setDescription("Indicator for Academic Job Title");
        metadata.setDisplayName("Has Academic Title");
        return metadata;
    }
}
